frontend/src/components/ui/GetEndpointButton.jsx

```diff
@@ -0,0 +1,67 @@
+import { useImmerAtom } from "jotai-immer";
+import axios from "axios";
+import { pipelineAtom } from "@/atoms/pipelineAtom";
+import { useMutation } from "@tanstack/react-query";
+import { Button } from "@carbon/react";
+import ClosableModal from "./modal/ClosableModal";
+import { useState } from "react";
+
+export default function GetEndpointButton({modalPopper, children, action}) {
+    const [pipeline, setPipeline] = useImmerAtom(pipelineAtom);
+    const [validationErrorMsg, setValidationErrorMsg] = useState([]);
+    const [isOpen, setIsOpen] = useState(false);
+
+    const styles = {
+        margin: '5px',
+    };
+
+    const mutation = useMutation({
+        mutationFn: async (uuid) => {
+            return axios.get(`${import.meta.env.VITE_EXECUTOR}/pipeline/${uuid}/list`)
+        },
+    })
+
+    const endpoint = async (pipeline) => {
+        const res = await mutation.mutateAsync(pipeline.id)
```

Comment:
I think this function call can throw an error, so you might want to wrap it in a try-catch